tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/SmokeTestsParser.java

```diff
@@ -0,0 +1,348 @@
+
+            //parse each test case
+            //operation name
+            SmokeTestData test = new SmokeTestData();
+            test.setOperationName(operationShape.getId().getName());
+
+            String inputShapeName = operationShape.getInput()
+                    .map(inputShape -> inputShape.getName().replaceAll("Input","Request") )
+                    .orElse(operationShape.getId().getName()+ "Request");
+
+            test.setInputShapeName(inputShapeName);
+
+            String outputShapeName = operationShape.getOutput()
+                    .map(outputShape -> outputShape.getName().replaceAll("Output","Result"))
+                    .orElse(operationShape.getId().getName()+ "Result");
+
+            test.setOutputShapeName(outputShapeName);
+            
+
+            //get params
+            if(testcase.getParams().isPresent())
+            {
+                Map<String, String> functionMap = new HashMap<String, String>();
+
+                List<String> functionOrder = new ArrayList<>();
+
+                //get input shape
+                Optional<Shape> topLevelShape = model.getShape(operationShape.getInput().get());
+
+                if(topLevelShape.isPresent()) {
+
+                    codegenAdapter.recordContainerForImport(topLevelShape.get());
+                    Map<String, Shape> fieldShapeMap = codegenAdapter.getMemberShapes(topLevelShape.get());
+
+                    //declare top level variable
+                    CppBlockWriter blockWriter = new CppBlockWriter(0)
+                        .addCode(String.format("%sRequest %s;\n", test.getOperationName(), "input"));
+            
+                    for (Map.Entry<String, Node> paramEntry : testcase.getParams().get().getStringMap().entrySet()) {
+                        String key = paramEntry.getKey();
+                        Shape fieldShape =  fieldShapeMap.get(key);
+                        if (fieldShape == null) {
+                            continue;
+                        }
+                        
+                        Node value = paramEntry.getValue();
+                        try {
+                            blockWriter.addCode(String.format("input.Set%s(%s);\n", GenericCodegenAdapter.capitalizeFirstLetter(key),
+
+                                codegenAdapter.GenerateCppSetters(
+                                        test.getOperationName().toLowerCase() + "_elem",
+                                        value,
+                                        fieldShape, //useful for C++ return type object
+                                        0, //useful for depth
+                                        0, //useful for array elements at same depth
+                                        functionMap,
+                                        functionOrder)
+                                ));
+                        } catch (RuntimeException e) {
+                            throw new RuntimeException(e);
```

Comment:
remove the try catch block completely since we dont need to catch to re-throw

Reply:
ok, that will mean first exception will cause exit. If that's indeed the idea, I will remove it

Reply:
yeah thats the idea, if we throw a exception we want to fail codegen, let the caller of codegen decide what to do after we know it has failed